### DISCUSSION TITLE
App-specific VPN connections

### DISCUSSION THREAD

**Discussion by anonbergine** [mullvad/mullvadvpn-app#9083] [Feature requests & Ideas]
## Issue report

* Windows 10
* Mullvad 2023.3

### Issue description

The split tunnelling feature allows me to define apps that _bypass_ VPN.

Rather than bypass VPN for specific app, I want to use a _different VPN endpoint_ for that app.

#### Use case:

* Default endpoint is in country **C**, to bypass (C)ensorship.
* I have an app that connects to a (S)erver in country **S**.
* Transit between **C** and **S** is slow / problematic / risky.

I realise I could do this with VMs, but tbh that's a PITA and I'm pretty sure I would screw things up and compromise my safety due to the fact I have no idea what I am doing (like, generally speaking, not just computers).

#### Scope Creep FTW:

App-specific endpoint support could potentially evolve to integrate with [Mozilla Container Tabs - VPN integration](https://support.mozilla.org/en-US/kb/protect-your-container-tabs-mozilla-vpn)? Would be super-useful for lightweight OSINT work.

**Comment by faern** [comment]
Thanks for the feature request. This would be quite hard to implement, and would require quite a lot of changes. We would need to change the server side also, since the same WireGuard key would then have an active tunnel to multiple servers, something we currently don't allow to prevent abuse. It would be hard to differentiate between a single user legitimately having multiple tunnels from multiple users just sharing the same WireGuard key to abuse the system.

**Comment by PhoneDroid** [comment]
*Unfortunate that this feature doesn't have a lot of*  
*traction yet , though I had to search a bit to even*  
*find it. ( Title needs some work )*

Not for the same reasons I would love to  
see this feature in some form as well.

A common use case would be using any multi-profile  
browser like Chromium or FireFox where each profile  
and it's corresponding browser instance uses a  
different VPN connection.

Not sure if anything changed about the challanges  
with the server side setup that MullVad is using,  
but wouldn't it be possible if users just cannot  
connect multiple tunnels to the same server?
